src/citrine/resources/generative_design_execution.py

```diff
@@ -0,0 +1,114 @@
+"""Resources that represent both individual and collections of design workflow executions."""
+import sys
+from typing import Optional, Union, Iterator
+from uuid import UUID
+
+from citrine._rest.collection import Collection
+from citrine._utils.functions import shadow_classes_in_module
+from citrine._session import Session
+import citrine.informatics.executions.generative_design_execution
+from citrine.informatics.executions.generative_design_execution import GenerativeDesignExecution
+from citrine.informatics.design_candidate import GenerativeDesignExecutionInput
+from citrine.resources.response import Response
+
+
+shadow_classes_in_module(
+    citrine.informatics.executions.generative_design_execution, sys.modules[__name__]
+)
+
+
+# TODO: Double check the path.
+class GenerativeDesignExecutionCollection(Collection["GenerativeDesignExecution"]):
+    """A collection of GenerativeDesignExecutions."""
+
+    _path_template = '/projects/{project_id}/generative-design/executions'
```

Comment:
I'm not sure if this is the correct path template? The base path for the API is `"api/v1/generative-design/executions"`, so I'm not sure if we need the project path explicitly. But @anoto-moniz  can confirm this.

Reply:
Actually I thnk I am wrong here. i think the public URI has the `projects` ID in the path, and only the proxy is shown above from backend.

Reply:
This was weird. When I had the "v1" in front of the expected path, my assertion was False. When I removed it (how you see) things worked again.

Reply:
If you want your test to check the path version, then you need to pass the `version` parameter to the `FakeCall` constructor. This way, if the caller doesn't care about the version, they're not forced to specify it in the path.

We really need to document the `FakeSession` and `FakeCall` classes...